### PR TITLE
Better get support (allow for parameters) 

### PR DESCRIPTION
**Pull request by mitchcapper**
I finished some initial work on ListOrders but I am aware you are not looking for object additions.   This however is the get method improvements I made to allow pagination support on my branch.

### CODE REVIEW

File: Source/Coinbase/CoinbaseApiAuthenticator.cs [hunk 1]
```diff
@@ -57,11 +58,15 @@ public void Authenticate(IRestClient client, IRestRequest request)
             var method = request.Method.ToString().ToUpper(CultureInfo.InvariantCulture);
 
             var body = string.Empty;
-
-            var param = request.Parameters.FirstOrDefault(p => p.Type == ParameterType.RequestBody);
-            if (param != null && param?.Value?.ToString() != "null" && !string.IsNullOrWhiteSpace(param?.Value?.ToString()))
-                body = param.Value.ToString();
-
+			if (request.Method != Method.GET) {
```
Comment (bchavez) [inline Source/Coinbase/CoinbaseApiAuthenticator.cs]: @mitchcapper , are you doubley sure we need this check against `Method.GET`? IIRC, `param` variable would be `null` if the HTTP method was already `Method.GET`.

Comment (mitchcapper) [inline Source/Coinbase/CoinbaseApiAuthenticator.cs]: No you are right it is not needed.  My goal was more to make it clear the case in which we were using PathAndQuery and leave the existing code relatively untouched.

Comment (bchavez) [inline Source/Coinbase/CoinbaseApiAuthenticator.cs]: Okay, no problem! Just wanted to double check. Thanks again for your contribution! One last question: Were you able to verify the code changes worked against sandbox/production?

Comment (mitchcapper) [inline Source/Coinbase/CoinbaseApiAuthenticator.cs]: I am using it in production I didn't do a test case on a bit of a crunch but my method that could be converted to a test case pretty easily:

public virtual CoinbaseResponse<Order[]> ListOrders(RequestPagination pagination = null) {  
 190   +			var args = new List<KeyValuePair<string,string>>();  
 191   +			if (pagination != null) {  
 192   +				if (!String.IsNullOrWhiteSpace(pagination.EndingBefore))  
 193   +					args.Add(new KeyValuePair<string, string>("ending_before", pagination.EndingBefore));  
 194   +                if (pagination.Limit != 0)  
 195   +					args.Add(new KeyValuePair<string, string>("limit", pagination.Limit.ToString()));  
 196   +				if (pagination.Order == SortOrder.Asc)  
 197   +					args.Add(new KeyValuePair<string, string>("order", "asc"));  
 198   +				if (!String.IsNullOrWhiteSpace(pagination.StartingAfter))  
 199   +					args.Add(new KeyValuePair<string, string>("starting_after", pagination.StartingAfter));  
 200   +            }  
 201   +			return SendGetRequest<Order[]>("orders",args.ToArray());  
 202   +		}

**Comment by bchavez** [comment]
Thanks again @mitchcapper . I really appreciate your contribution. Do you need a release now? or do you have more changes coming?

**Comment by mitchcapper** [comment]
No need for a release I am using my own version anyway (happy to have the object models for my use).  If I find more generics I can do a PR for as I go I will do so. Thanks for the assistance and a good lib:)

**Comment by waxmihs2902** [comment]
thanks!

**Pull request merged**